dispatcher/dispatcher_test.go

```diff
@@ -0,0 +1,221 @@
+package dispatcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type testInterface1 interface {
+	TestFunc1()
+}
+
+type testInterface2 interface {
+	TestFunc2()
+}
+
+type testEvent1 struct {
+}
+
+type testEvent2 struct {
```

Comment:
should be called *Spy, see https://8thlight.com/blog/uncle-bob/2014/05/14/TheLittleMocker.html